src/sinks/vector/v2/sink.rs

```diff
@@ -0,0 +1,56 @@
+use crate::event::proto::EventWrapper;
+use crate::event::{Event, EventFinalizers, Finalizable};
+use crate::sinks::util::{SinkBuilderExt, StreamSink};
+use crate::sinks::vector::v2::service::{VectorRequest, VectorResponse};
+use crate::Error;
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use prost::Message;
+
+use tower::util::BoxService;
+use vector_core::buffers::Acker;
+use vector_core::stream::{batcher, BatcherSettings};
+use vector_core::ByteSizeOf;
+
+struct EventData {
+    byte_size: usize,
+    finalizers: EventFinalizers,
+    wrapper: EventWrapper,
+}
+
+pub struct VectorSink {
+    pub batch_settings: BatcherSettings,
+    pub service: BoxService<VectorRequest, VectorResponse, Error>,
```

Comment:
We should have this be generic so we can avoid the box.

Reply:
For some reason I thought something required the `Box`ed version. I'll try to change it.

Reply:
```rust
let service = ServiceBuilder::new()
            .settings(request_settings, VectorGrpcRetryLogic)
            .service(service);
```

This returns a `BoxService`. I can make it generic so maybe we can remove that in the future, but it's not going to do anything useful now. (The actual boxing is in `sinks/util/service` line 310 if you are curious)

Reply:
I'm going to make an issue to remove the `BoxService` from the layer, and we can change this in all the sinks at once.

Reply:
https://github.com/vectordotdev/vector/issues/10158

Reply:
Yeah, I guess I was generic-ifying stuff without ever actually realizing it was boxed anyways.  TIL!